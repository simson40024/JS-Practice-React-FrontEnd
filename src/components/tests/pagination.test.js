import Pagination from "../shopping/pagination";
import { createRoot } from "react-dom/client";
import { act } from "@testing-library/react";
import * as hooks from "../../hook";

let container = null;
beforeEach(() => {
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  container.remove();
  container = null;
});

const state = {
  pagination: {
    currentPage: 2,
    limitOnPage: 10,
    total: 4,
  },
};

const mockSelector = jest.spyOn(hooks, "useAppSelector");
const mockDispatch = jest.spyOn(hooks, "useAppDispatch");

describe("pagination", () => {
  it("shoud create Pagination", () => {
    mockSelector.mockImplementation((selector) => selector(state));
    mockDispatch.mockReturnValue(jest.fn());
    act(() => {
      createRoot(container).render(<Pagination />);
    });
    expect(container).toMatchSnapshot();
  });
});
