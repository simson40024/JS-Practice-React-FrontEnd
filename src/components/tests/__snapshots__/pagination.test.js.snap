// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pagination shoud create Pagination 1`] = `
<div>
  <div
    class="pagination"
  >
    <button>
      &lt;
    </button>
    <input
      id="page-number"
      value="2"
    />
    <button>
      &gt;
    </button>
  </div>
</div>
`;
